pydantic/_internal/_model_construction.py

```diff
@@ -307,4 +310,6 @@ def apply_alias_generator(config: ConfigDict, fields: dict[str, FieldInfo]) -> N
             if not isinstance(alias, str):
                 raise TypeError(f'alias_generator {alias_generator} must return str, not {alias.__class__}')
             field_info.alias = alias
+            field_info.validation_alias = alias
+            field_info.serialization_alias = alias
```

Comment:
serialization alias should only be set here if `alias` is a string, we need a test for this.

Reply:
based on the error above, `alias_generator` only returns string

Reply:
as `alias_generator` always returns a string this check is not needed.